core/src/main/scala/kafka/server/ReplicaAlterLogDirsThread.scala

```diff
@@ -127,6 +131,19 @@ class ReplicaAlterLogDirsThread(name: String,
     logAppendInfo
   }
 
+  override def addPartitions(initialFetchStates: Map[TopicPartition, OffsetAndEpoch]): Set[TopicPartition] = {
+    partitionMapLock.lockInterruptibly()
+    try {
+      // It is possible that the log dir fetcher completed just before this call, so we
+      // filter only the partitions which still have a future log dir.
+      val filteredFetchStates = initialFetchStates.filterKeys(replicaMgr.futureLogExists)
+      super.addPartitions(filteredFetchStates)
```

Comment:
Should we skip to create thread if the collection of filtered states is empty?

Reply:
I think this will be a no-op in `addPartitions`, so it's probably harmless